subsys_read_user_input.py

```diff
@@ -36,96 +38,114 @@ class ModeStatus:
+                gp_map = [gp['name'] for gp in Gamepad.map_list].index('Default')
+                cls.joystick_maps.append(Gamepad.map_list[gp_map])
+            joystick.init()
 
     @classmethod
     def run(cls, rc_threshold: int) -> (type(RCStatus), type(KeyStatus), type(ModeStatus)):
         for event in pygame.event.get():
             if event.type == pygame.QUIT:
                 RunStatus.value = RUN.STOP
-            elif event.type == pygame.KEYDOWN:
-                KeyStatus.is_pressed = True
-                KeyStatus.type_pressed = event.key
-                if event.key == pygame.K_ESCAPE:
-                    RunStatus.value = RUN.STOP
-                elif event.key == pygame.K_SPACE:
-                    ModeStatus.value = MODE.EMERGENCY
-                elif event.key == pygame.K_t:
-                    ModeStatus.value = MODE.TAKEOFF
-                elif event.key == pygame.K_l:
-                    ModeStatus.value = MODE.LAND
-                else:
-                    cls.__key_down(event.key, rc_threshold)
-            elif event.type == pygame.KEYUP:
+            elif event.type == pygame.JOYAXISMOTION:
+                try:
+                    axis = cls.joystick_maps[event.joy]['axes'][event.axis]
+                    cls.axis_motion(axis, event.value, rc_threshold)
+                except KeyError:
+                    print('No axis found with index %i in the Gamepad map', event.axis)
+            elif event.type == pygame.JOYBUTTONDOWN:
+                try:
+                    KeyStatus.is_pressed = True
+                    button = cls.joystick_maps[event.joy]['buttons'][event.button]
+                    KeyStatus.type_pressed = button
+                    cls.buttons(button, rc_threshold, KeyStatus)
+                except KeyError:
+                    print('No button found with index %i in the Gamepad map', event.button)
+            elif event.type == pygame.JOYBUTTONUP:
                 KeyStatus.is_pressed = False
                 KeyStatus.type_pressed = None
-                cls.__key_up(event.key)
+            elif event.type == pygame.KEYDOWN:
+                try:
+                    KeyStatus.is_pressed = True
+                    button = Gamepad.keyboard_map['buttons'][event.key]
+                    KeyStatus.type_pressed = button
+                    cls.buttons(button, rc_threshold, KeyStatus)
+                except KeyError:
+                    print('No key found with index %i in the keyboard map', event.key)
+            elif event.type == pygame.KEYUP:
+                try:
+                    KeyStatus.is_pressed = False
+                    button = Gamepad.keyboard_map['buttons'][event.key]
+                    KeyStatus.type_pressed = None
+                    cls.buttons(button, rc_threshold, KeyStatus)
+                except KeyError:
+                    print('No key found with index %i in the keyboard map', event.key)
```

Comment:
Instead of printing a generic error message + the key which caused the error you might want not to do this individually for similar cases like that.

=> remove try except and encapsulate the whole IF block with in a try except. You should be able to display the problematic key using the property of the KeyError (which is an Exception/Error/Instance object that should contain all the traceback information you need [assuming the *.py which raise the KeyError was programmed correctly])

See also : https://docs.python.org/3/tutorial/errors.html